### ISSUE TITLE
logging_is_not_google_glog.h:24:11: error: 'const int ERROR' redeclared as different kind of symbol  const int ERROR = 2;            ^

### ISSUE THREAD

**Issue by ksasso1028** [pytorch/pytorch#37629]
I have a file in my large project using ERROR as a constant already globally so i cannot change that variable as its used in many places.  When i try to build i get an error from the linker because it finds both variables, whats the best way to edit "logging_is_not_google_glog.h" without breaking anything else? Anyway i could remove it entirely?

**Comment by ksasso1028** [comment]
I am building pytorch from source , v1.3.0 . Any way i can modify the build so there is no issues?

**Comment by ksasso1028** [comment]
Would using google log eliminate this ?

**Comment by malfet** [comment]
@ksasso1028 no, using google log would not help, since it defines `ERROR` in global namespace as well, see https://github.com/google/glog/blob/master/src/glog/log_severity.h#L69

But it shouldn't be a problem, unless you are linking statically with pytorch, since `ERROR` should be hidden (at least 1.4.0 by default compiles the library with `-fvisibility=hidden` which should prevent any symbol not defined with TORCH_API prefix  being visible from main application.)

Bad ways of fixing the duplicated symbols are to:
 - put log severely constants into anonymous namespace
 - Replace `const int ERROR = 2` with 
```
#ifndef ERROR
#define ERROR 2
#endif
```
A much better way would be:
 - Put those constants into its own namespace
 - Defining it as `static constexpr int ERROR = 2;` (if compiler is smart enough it should not emit any symbols
